components/recipe_table_comp.js

```diff
@@ -0,0 +1,113 @@
+import { Button, Chip } from '@mui/material';
+import StarIcon from '@mui/icons-material/Star'
+import StarBorderIcon from '@mui/icons-material/StarBorder'
+import React, { Component } from 'react';
+import View_Recipe from "./view_recipe"
+import { DataGrid } from '@mui/x-data-grid'
+import { Box } from '@mui/system';
+
+class RecipeTable extends Component {
+	state = { 
+        recipeOpen: false,
+        currRecipe: this.props.recipes[0],
+        shownRecipes: this.formatRecipes(this.props.recipes)
+    }
+
+    cols = [
+        { field: 'key', headerName: 'View Recipe', width: 100, renderCell: (params) => {
+            return <Button variant='contained' onClick={() => this.handleToggleRecipe(params.value)}>View</Button>
+        }},
+        { field: 'title', headerName: 'Recipe Name', width: 300},
+        { field: 'time', headerName: 'Time', width: 80, valueFormatter: (params) => {
+            return '' + params.value + " mins"
+        }},
+        { field: 'difficulty', headerName: 'Difficulty', width: 140, renderCell: (params) => {
+            return this.getStarsFromNum(params.value)
+        }},
+        { field: 'rating', headerName: 'Rating', width: 140, renderCell: (params) => {
+            return this.getStarsFromNum(params.value)
+        }},
+        { field: 'tags', headerName: 'Tags', minWidth: 300, renderCell: (params) => {
+            return this.getTags(params.value)
+        }}
+    ]
+
+    formatRecipes(recipes) {
+        recipes.forEach((recipe) => {
+            recipe.key = recipe.id
+            recipe.rating = this.getAverageRating(recipe.reviews)
+            recipe.difficulty = this.getAverageDifficulty(recipe.reviews)
+        })
+        return recipes
+    }
+
+    getAverageDifficulty(reviews) {
+        let sum = 0
+        Object.values(reviews).forEach( review => {
+            sum += review.difficulty;
+        })
+        return (sum / reviews.length).toFixed(1)
+    }
+    getAverageRating(reviews) {
+        let sum = 0
+        Object.values(reviews).forEach( review => {
+            sum += review.rating;
+        })
+        return (sum / reviews.length).toFixed(1)
+    }
```

Comment:
Same as above, but these may not need to be separate functions, since the logic is identical, you could even pass the field name as an argument an then access that field, so 
getAverage(reviews, field){
... sum += review[field]
}